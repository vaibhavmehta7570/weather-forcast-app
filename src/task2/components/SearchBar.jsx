import React from 'react'

const SearchBar = ({ input, onCityChange, onSubmit, statusCode }) => {
  const getSearchBarClasses = () => {
    let classes = 'searchbar__input'
    return (classes += statusCode !== 200 ? ' searchbar__input--error' : '')
  }

  return (
    <form className='searchbar' onSubmit={(e) => onSubmit(e)}>
      <input
        type='text'
        placeholder='Enter city'
        className={getSearchBarClasses()}
        value={input}
        onChange={onCityChange}
      />
      <button type='submit' className='searchbar__submit'>
        find!
      </button>
    </form>
  )
}

export default SearchBar
