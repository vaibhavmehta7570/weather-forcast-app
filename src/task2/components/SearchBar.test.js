import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { configure, shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import SearchBar from './SearchBar'
configure({ adapter: new Adapter() })

describe('Should test SearchBar.jsx', () => {
  test('should render correctly Searchbar component', () => {
    const rendererResponse = renderer.create(<SearchBar />)
    expect(rendererResponse).toMatchSnapshot()
  })

  test('should submit form correctly', () => {
    const fakeEvent = {
      preventDefault: () => 'preventDefault called',
      stopPropagation: () => 'stopPropagation called'
    }
    const mockOnSubmit = jest.fn(() => 'hello')
    const SearchBarWrapper = shallow(<SearchBar onSubmit={mockOnSubmit} />)
    const submitForm = SearchBarWrapper.find('.searchbar')
    expect(submitForm.length).toBe(1)
    submitForm.simulate('submit', fakeEvent)
  })
})
