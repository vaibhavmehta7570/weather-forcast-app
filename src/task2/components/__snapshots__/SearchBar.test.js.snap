// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test SearchBar.jsx should render correctly Searchbar component 1`] = `
<form
  className="searchbar"
  onSubmit={[Function]}
>
  <input
    className="searchbar__input searchbar__input--error"
    placeholder="Enter city"
    type="text"
  />
  <button
    className="searchbar__submit"
    type="submit"
  >
    find!
  </button>
</form>
`;
